components/scream/src/physics/p3/p3_functions_f90.hpp

```diff
@@ -1037,13 +1037,14 @@ void calculate_incloud_mixingratios_f(Real qc, Real qr, Real qi, Real qm, Real n
 
 struct P3MainPart1Data : public PhysicsTestData
 {
-  static constexpr size_t NUM_ARRAYS = 39;
+  static constexpr size_t NUM_ARRAYS = 40;
 
```

Comment:
We have the nccn_prescribed data, see below @PeterCaldwell 

struct P3MainPart1Data : public PhysicsTestData
{
  static constexpr size_t NUM_ARRAYS = 40;

  // Inputs
  Int kts, kte, kbot, ktop, kdir;
  bool do_predict_nc, do_prescribed_CCN;
  Real dt;
  Real* pres, *dpres, *dz, *nc_nuceat_tend, *exner, *inv_exner, *inv_cld_frac_l, *inv_cld_frac_i, *inv_cld_frac_r, *latent_heat_vapor, *latent_heat_sublim,
     *latent_heat_fusion, *nccn_prescribed;

Reply:
Ok, you're right. I got confused because you added 2 new variables but only increased NUM_ARRAYS by 1. I see now that is because do_prescribed_CCN isn't an array, but rather a scalar.